### DISCUSSION TITLE
How can I refresh the auth

### DISCUSSION THREAD

**Discussion by rApeNB** [mkb79/Audible#120] [Q&A] [answered]
Hello

When I load the authentication information from the file, generally speaking, it has expired. How do I re-authenticate. I didn't find the corresponding api.

![e87b1a67dea0b57388f05fbdb11b299d](https://user-images.githubusercontent.com/1839378/229718750-740c67b8-92a0-48ac-8190-b2d4b11153ca.png)
At this point, I can't use auth to make any requests.

**Comment by mkb79** [comment, marked as answer]
Hi.

By default my package make use of the message authentication code (MAC) to authenticate the client to the Audible server. Using MAC for authorization does not need an access token. A valid token is therefore unimportant. 

The only point where an valid access token matters is on deregistration part. But `auth.deregister_device()` will refresh the access token automatically before unregister a device. 

But to answer your question. To manually update an access token you can call `auth.refresh_access_token()`. This will update the token only if they are expired. To force a refresh call `auth.refresh_access_token(force=True)`.

**Reply by rApeNB** [reply]
Thanks!
![c5795999857f09909e72dd1fff47819b](https://user-images.githubusercontent.com/1839378/229732681-a59b095e-64c3-4d07-8979-0ccc3eea7dfb.png)
When I call it, it reports an error directly. What is the reason for this, please?

**Reply by mkb79** [reply]
It seems you have an invalid refresh token. Do you have deregister your device before? Or you have removed the device from the Amazon page?

You can try to make a request like `client.get("library")` and make sure you got a valid response. If your credentials are revoked you are get a status code in 4XX range. 

Edit:
Or try to register a new device

**Reply by rApeNB** [reply]
Thank you very much for your answer. I think I logged out of the device information causing this problem.

**Reply by mkb79** [reply]
> Thank you very much for your answer. I think I logged out of the device information causing this problem.

If you need any help using my package feel free to contact me. I'll help you (if I can)!
